clar2wasm/src/words/contract.rs

```diff
@@ -68,6 +71,31 @@ impl ComplexWord for ContractCall {
                 .i32_const(id_length as i32);
         } else {
             // This is a dynamic contract call (via a trait).
+            // Push the trait name on the stack
+            let dynamic_arg = contract_expr.match_atom().ok_or_else(|| {
+                GeneratorError::TypeError(
+                    "Dynamic contract-call? argument should be a name".to_owned(),
+                )
+            })?;
+            let trait_signature = generator
+                .get_current_function_arg_type(dynamic_arg)
```

Comment:
What if the trait arg's name is a local binding:
```clarity
(define-public (foo (t <my-trait>))
  (let ((t1 t))
    (contract-call? t1 bar)
  )
)
```

Reply:
f